src/main/scala/org/ergoplatform/api/routes/InfoRoute.scala

```diff
@@ -39,38 +43,65 @@ case class InfoRoute(readersHolder: ActorRef,
-    val stateVersion = readers.s.map(_.version).map(Algos.encode)
-    val bestHeader = readers.h.flatMap(_.bestHeaderOpt)
-    val bestFullBlock = readers.h.flatMap(_.bestFullBlockOpt)
-    val unconfirmedCount = readers.m.map(_.size).getOrElse(0)
-    val stateRoot = readers.s.map(s => Algos.encode(s.rootHash)).getOrElse("null")
-    Map(
-      "name" -> Algos.encode(nodeId).asJson,
-      "appVersion" -> Version.VersionString.asJson,
-      "headersHeight" -> bestHeader.map(_.height).getOrElse(0).asJson,
-      "fullHeight" -> bestFullBlock.map(_.header.height).getOrElse(0).asJson,
-      "bestHeaderId" -> bestHeader.map(_.encodedId).getOrElse("null").asJson,
-      "bestFullHeaderId" -> bestFullBlock.map(_.header.encodedId).getOrElse("null").asJson,
-      "previousFullHeaderId" -> bestFullBlock.map(_.header.parentId).map(Base58.encode).getOrElse("null").asJson,
-      "stateRoot" -> stateRoot.asJson,
-      "difficulty" -> bestFullBlock.map(_.header.requiredDifficulty).getOrElse(BigInt(0)).asJson,
-      "unconfirmedCount" -> unconfirmedCount.asJson,
-      "stateType" -> stateType.asJson,
-      "stateVersion" -> stateVersion.asJson,
-      "isMining" -> minerInfo.isMining.asJson,
-      "votes" -> Algos.encode(minerInfo.votes).asJson,
-      "peersCount" -> connectedPeersLength.asJson
-    ).asJson
+
+  case class Info(
+                   name: String,
+                   version: String,
+                   headersHeight: Int,
+                   fullHeight: Int,
+                   bestHeaderId: Option[String],
+                   bestFullHeaderId: Option[String],
+                   previousFullHeaderId: Option[String],
+                   stateRoot: Option[String],
+                   difficulty: Long,
+                   unconfirmedCount: Int,
+                   stateType: String,
+                   stateVersion: Option[String],
+                   isMining: Boolean,
+                   votes: String,
+                   peersCount: Int
+                 )
+
+  object Info {
+
+    def apply(nodeId: Array[Byte],
+              minerInfo: MiningStatusResponse,
+              connectedPeersLength: Int,
+              readers: Readers,
+              stateType: String): Info = {
+      val stateVersion = readers.s.map(_.version).map(Algos.encode)
+      val bestHeader = readers.h.flatMap(_.bestHeaderOpt)
+      val bestFullBlock = readers.h.flatMap(_.bestFullBlockOpt)
+      val unconfirmedCount = readers.m.map(_.size).getOrElse(0)
+      val stateRoot = readers.s.map(s => Algos.encode(s.rootHash))
+      Info(
+        Algos.encode(nodeId),
+        Version.VersionString,
+        bestHeader.map(_.height).getOrElse(0),
+        bestFullBlock.map(_.header.height).getOrElse(0),
+        bestHeader.map(_.encodedId),
+        bestFullBlock.map(_.header.encodedId),
+        bestFullBlock.map(_.header.parentId).map(Base58.encode),
+        stateRoot,
+        bestFullBlock.map(_.header.requiredDifficulty).getOrElse(BigInt(0)).toLong,
```

Comment:
Is it safe to do ".toLong" here? Difficulty may be much bigger then long

Reply:
I didn't know that. So how we can render this kind of numbers with circe? Maybe it will be better to expect all kinds of number in response? By default `BigInt(10).asJson` will be rendered as `1e1`.

Reply:
rendering it as `String` ? it's only `info` endpoint - shouldn't be used as source of data for any important computations. and even if so - it's not difficult to parse it to `BigInt` on client side.